### DISCUSSION TITLE
Tpot on Microsoft Azure Help

### DISCUSSION THREAD

**Discussion by jdb71380** [telekom-security/tpotce#1051] [Q&A] [answered]
Hi there,

I was wondering if anyone has had success running Tpot on Microsoft Azure. I have successfully run Tpot in VMWare and was wanting to start hosting it on Microsoft Azure Cloud. I got it running (for the most part) with a few errors. Was wondering if anyone knew how to solve this issue. 
![TpotDashboardSnip](https://user-images.githubusercontent.com/97466690/160893480-961d88c4-d5c7-400b-aa3d-3fb69b7a1cf2.PNG)
![TpotKibanaSnip](https://user-images.githubusercontent.com/97466690/160893481-256c350c-e04f-41c0-a872-f6badc5c446a.PNG)

As you can see the Kibana dashboard is throwing some errors. Along with all the individual dashboards in Kibana. I checked to make sure I have 8GB of memory and a large enough hard drive. I also took at look at the Kibana documentation, but it wasn't to helpful. The admin portal on port 64294 works correctly and the terminal also seems to be working fine. Any ideas why Kibana is throwing a fit?

**Comment by t3chn0m4g3** [comment]
This is actually how it should look right after the installation since no data has been pushed from `logstash` to `elasticsearch`.

You can try to comment the following lines in `/etc/sysctl.conf`:
```
#net.ipv6.conf.all.disable_ipv6 = 1
#net.ipv6.conf.default.disable_ipv6 = 1
#net.ipv6.conf.lo.disable_ipv6 = 1
```

But it is far more interesting to see the logs ...
```
docker logs logstash
docker logs elasticsearch
```

Also check the logs in `/install.log` and `/install.err`.

T-Pot's cloud support is developed solely by the community. Some users report success on AWS, GCP and even Azure.
However the best supported Cloud Plattform is Telekom OTC.

**Comment by jdb71380** [comment]
Thank you for the reply,

I went ahead and tried going to port 80 to create a log. It didn't seem to change anything on Kibana. It still looked the exact same. 

I also checked the docker logs logstash as you suggested. This is what most look like. Is this abnormal? It seems to be throwing an error about being unable to commit memory. But as you can see I have 8gB of memory.
![image](https://user-images.githubusercontent.com/97466690/161322275-1950288d-2fd6-47a2-b2a1-ed613baba470.png)

Any further advice?

**Reply by t3chn0m4g3** [reply]
```
docker exec -it logstash ash
ps -ef | grep Xms
```

Check if the output has two Xms and Xmx options with more than 1G.

If not you must set / increase Java Opts for RAM in `/opt/tpot/etc/tpot.yml`.

**Comment by jdb71380** [comment, marked as answer]
Thank you for the reply,

I just wanted to update this. It was definitely that I did not have enough RAM to be running all those services. I went ahead and deleted and remade my T-pot VM in Azure. This time I created it with 16 Gb of RAM.  
![image](https://user-images.githubusercontent.com/97466690/161601203-44295d31-87ce-4b4e-9b4d-7c81c110f3a9.png)
It seems to be running at around 12 Gb of RAM usage steady. So anyone wanting to use Azure to create T-pot it is possible, but might require more resources than the recommended.
